Nos/Models/RelaySubscription.swift

```diff
@@ -36,4 +39,54 @@ struct RelaySubscription: Identifiable {
     var isOneTime: Bool {
         filter.limit == 1
     }
+    
+    internal init(
+        filter: Filter, 
+        relayAddress: URL, 
+        subscriptionStartDate: Date? = nil, 
+        oldestEventCreationDate: Date? = nil, 
+        referenceCount: Int = 0
+    ) {
+        self.filter = filter
+        self.relayAddress = relayAddress
+        self.id = (filter.id + "-" + relayAddress.absoluteString).sha256()
+        self.subscriptionStartDate = subscriptionStartDate
+        self.oldestEventCreationDate = oldestEventCreationDate
+        self.referenceCount = referenceCount
+    }
 }
+
+/// A handle that holds references to one or more `RelaySubscription`s and provides the ability to cancel these 
+/// subscriptions. Will auto-cancel them when it is deallocated. Modeled after Combine's `Cancellable`.
+class SubscriptionCancellable {
```

Comment:
Can you move this class to its own file?

Reply:
👍